src/liveEditor/generators/generateToolbar.tsx

```diff
@@ -54,16 +56,17 @@ export function appendFieldPathDropdown(
     const { editableElement: targetElement, fieldMetadata } = eventDetails;
     const targetElementDimension = targetElement.getBoundingClientRect();
 
-    const distanceFromTop =
-        targetElementDimension.top +
-        window.scrollY -
-        LIVE_PREVIEW_OUTLINE_WIDTH_IN_PX -
-        5;
+    const distanceFromTop = targetElementDimension.top + window.scrollY - TOOLBAR_EDGE_BUFFER;
+    const adjustedDistanceFromTop = targetElementDimension.top < TOP_EDGE_BUFFER
```

Comment:
```suggestion
    const adjustedDistanceFromTop = targetElementDimension.top + window.scrollY < TOP_EDGE_BUFFER
```
I think this will ensure that when there is some scroll space, we still draw it on top. Only if there is no space at all, usually when it's the topmost visible item on the page, we draw it on bottom

Reply:
As discussed, change works for now. We can clarify more and add this refinement later